lib/parse_tree.rb

```diff
@@ -0,0 +1,85 @@
+require_relative 'parse_tag'
+
+Tag = Struct.new(:type, :classes, :id, :name, :src, :title, :text, :children, :parent)
```

Comment:
I ran into trouble having text as an attribute of Tag. I tried to make it a node of its own, but I couldn't quite get that to work in time.